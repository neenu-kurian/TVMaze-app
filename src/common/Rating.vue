<template>
  <div v-if="rating" class="show__rating--wrapper">
    <img
      src="../assets/images/star-regular.png"
      alt="rating"
      class="show__rating--icon"
    />
    <span class="show__rating">{{ rating }}</span>
  </div>
</template>

<script>
export default {
  name: 'Rating',
  props: {
    rating: Number
  }
}
</script>
