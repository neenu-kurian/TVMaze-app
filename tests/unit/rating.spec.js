import { mount } from '@vue/test-utils'
import Rating from '@/common/Rating'

it('displays rating correctly', () => {
  const wrapper = mount(Rating, {
    propsData: {
      rating: 8
    }
  })

  expect(wrapper.text()).toContain('8')
})

it('renders the component correctly', () => {
  const wrapper = mount(Rating, {
    propsData: {
      rating: 10
    }
  })

  expect(wrapper).toMatchSnapshot()
})

it('doesnt render the rating if not present', () => {
  const wrapper = mount(Rating)

  expect(wrapper.find('.show__rating--wrapper').exists()).toBe(false)
})
