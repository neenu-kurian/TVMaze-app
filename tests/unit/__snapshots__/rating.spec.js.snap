// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnt render the rating correctly 1`] = `<div class="show__rating--wrapper"><img src="../assets/images/star-regular.png" alt="rating" class="show__rating--icon"> <span class="show__rating">10</span></div>`;

exports[`doesnt render the rating if not present 1`] = `<div class="show__rating--wrapper"><img src="../assets/images/star-regular.png" alt="rating" class="show__rating--icon"> <span class="show__rating">10</span></div>`;

exports[`doesnt renders the rating correctly 1`] = `<div class="show__rating--wrapper"><img src="../assets/images/star-regular.png" alt="rating" class="show__rating--icon"> <span class="show__rating">10</span></div>`;

exports[`renders the component correctly 1`] = `<div class="show__rating--wrapper"><img src="../assets/images/star-regular.png" alt="rating" class="show__rating--icon"> <span class="show__rating">10</span></div>`;

exports[`renders the component correctly 2`] = `<div class="show__rating--wrapper"><img src="../assets/images/star-regular.png" alt="rating" class="show__rating--icon"> <span class="show__rating">10</span></div>`;

exports[`renders the rating correctly 1`] = `<div class="show__rating--wrapper"><img src="../assets/images/star-regular.png" alt="rating" class="show__rating--icon"> <span class="show__rating">10</span></div>`;
